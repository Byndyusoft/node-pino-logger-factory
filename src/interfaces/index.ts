export * from "./pino-http-logger-factory.interface";
export * from "./pino-http-logger-options-builder.interface";
export * from "./pino-logger-factory.interface";
export * from "./pino-logger-options-builder.interface";
