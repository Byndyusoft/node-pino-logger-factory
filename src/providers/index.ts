export * from "./pino-http-logger-factory";
export * from "./pino-http-logger-options-builder";
export * from "./pino-logger-factory";
export * from "./pino-logger-options-builder";
