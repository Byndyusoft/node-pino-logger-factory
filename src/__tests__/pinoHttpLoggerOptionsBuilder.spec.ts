/*
 * Copyright 2021 Byndyusoft
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { IncomingMessage, ServerResponse } from "http";

import { Span } from "opentracing";
import { AutoLoggingOptions } from "pino-http";

import { PinoHttpLoggerOptionsBuilder, PinoLoggerFactory } from "~/src";

describe("PinoHttpLoggerOptionsBuilder", () => {
  it("must build options with defaults", () => {
    const builder = new PinoHttpLoggerOptionsBuilder();

    expect(builder.build()).toMatchSnapshot();
  });

  it("must build options without defaults", () => {
    const builder = new PinoHttpLoggerOptionsBuilder(false);

    expect(builder.build()).toMatchSnapshot();
  });

  it("must override logger", () => {
    const logger = new PinoLoggerFactory().create();

    const options = new PinoHttpLoggerOptionsBuilder()
      .withLogger(logger)
      .build();

    expect(options.logger).toBe(logger);
  });

  it("must return path", () => {
    const options = new PinoHttpLoggerOptionsBuilder().build();

    expect(
      (options.autoLogging as AutoLoggingOptions).getPath?.({
        originalUrl: "/test",
      } as unknown as IncomingMessage),
    ).toStrictEqual("/test");
  });

  it("must set traceId if __rootSpan__ in request", () => {
    const span = new Span();
    jest.spyOn(span.context(), "toTraceId").mockReturnValue("120a073f7c3954d5");

    const options = new PinoHttpLoggerOptionsBuilder().build();

    expect(
      options.reqCustomProps?.(
        { __rootSpan__: span } as unknown as IncomingMessage,
        {} as unknown as ServerResponse,
      ),
    ).toStrictEqual({
      traceId: "120a073f7c3954d5",
    });
  });

  it("must set traceId to undefined if __rootSpan__ not in request", () => {
    const options = new PinoHttpLoggerOptionsBuilder().build();

    expect(
      options.reqCustomProps?.(
        {} as unknown as IncomingMessage,
        {} as unknown as ServerResponse,
      ),
    ).toStrictEqual({
      traceId: undefined,
    });
  });
});
