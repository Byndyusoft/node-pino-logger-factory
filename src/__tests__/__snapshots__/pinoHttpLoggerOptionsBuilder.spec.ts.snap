// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinoHttpLoggerOptionsBuilder must build options with defaults 1`] = `
{
  "autoLogging": {
    "ignore": [Function],
  },
  "customProps": [Function],
  "logger": undefined,
  "wrapSerializers": false,
}
`;

exports[`PinoHttpLoggerOptionsBuilder must build options without defaults 1`] = `
{
  "autoLogging": {
    "ignore": [Function],
  },
  "customProps": [Function],
  "logger": undefined,
  "wrapSerializers": false,
}
`;
